manager/src/main/java/com/redhat/service/bridge/manager/BridgeWorker.java

```diff
@@ -101,17 +124,45 @@ public void step1_createKafkaTopic(Bridge bridge) {
         transact(bridge);
     }
 
+    @Transactional
+    @ConsumeEvent(value = DELETE_KAFKA_TOPIC, blocking = true)
+    public void step1_deleteKafkaTopic(Bridge bridge) {
+        try {
+            rhoasService.deleteTopicAndRevokeAccessFor(internalKafkaConfigurationProvider.getBridgeTopicName(bridge), RhoasTopicAccessType.CONSUMER_AND_PRODUCER);
+        } catch (RuntimeException e) {
+            LOGGER.error("Failed to create the topic. The retry will be performed by the reschedule loop.", e);
+            PreparingWorker preparingWorker = preparingWorkerDAO.findByEntityId(bridge.getId()).get(0);
+            preparingWorker.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
+            preparingWorker.setStatus(BridgeWorkerStatus.FAILURE.toString()); // where you are. The rescheduler will fire the event according to the desiredStatus
+            transact(preparingWorker);
+            return;
+        }
+
+        PreparingWorker preparingWorker = preparingWorkerDAO.findByEntityId(bridge.getId()).get(0);
+        preparingWorkerDAO.deleteById(preparingWorker.getId());
+
+        // The bridge is ready to be provisioned by the shard.
+        bridge.setStatus(ManagedEntityStatus.DEPROVISION);
+        bridge.setDesiredStatus(ManagedEntityStatus.DELETING);
```

Comment:
Isn't the desired status `DELETED`? 

Or does each step set the desired state to be what it should be for when the step completes successfully.

Reply:
I still have to think about this. I'll get back to you

Reply:
Investigating an issue I have here with my MC changes; I think having each step setting the desired status to what it should be at the end of the step is better (than what I wrote above suggesting it should be the terminal/ultimate status). The problem I have here with using the terminal/ultimate status is that the re-try mechanism is not able to ascertain what the next step should be - when you have multiple steps in a chain as we do for MC provisioning.

Reply:
tl; dr; I think you are correct.